### PR TITLE
docs: Python Agent v9.5.0 Release

### PR DESCRIPTION
**Pull request by TimPansino**
## Give us some context

* Add release notes for Python agent v9.5.0
* Update settings and API documentation for changes in this version

### CODE REVIEW

**Comment by TimPansino** [comment]
Would be very helpful to get this out ASAP since we did a release at the end of our day. If you have any suggestions feel free to commit them and merge this, and I can open a follow up PR later if I have anything to address.

File: src/content/docs/apm/agents/python-agent/configuration/python-agent-configuration.mdx [hunk 1]
```diff
@@ -3950,6 +3950,175 @@ For some tips on configuring logs for the Python agent, see [Configure Python lo
...
+          </th>
+
+          <td>
+            `NEW_RELIC_APPLICATION_LOGGING_FORWARDING_CONTEXT_DATA_INCLUDE`
+          </td>
+        </tr>
+      </tbody>
+    </table>
+
+    If attributes are enabled for context_data, all attribute keys found in this list will be sent to us in transaction traces. For more information, see the [agent attribute rules](/docs/apm/other-features/attributes/agent-attributes).
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/python-agent/configuration/python-agent-configuration.mdx:4054]:
```suggestion
    If attributes are enabled for `context_data`, all attribute keys found in this list will be sent to us in transaction traces. For more information, see the [agent attribute rules](/docs/apm/other-features/attributes/agent-attributes).
```

File: src/content/docs/apm/agents/python-agent/configuration/python-agent-configuration.mdx [hunk 2]
```diff
@@ -3950,6 +3950,175 @@ For some tips on configuring logs for the Python agent, see [Configure Python lo
...
+
+    <Callout variant="important">
+      When adding context attributes, the agent prefixes the keys with `context.` for attributes from the logging framework context, and `message.` for attributes from a dictionary message context.
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/python-agent/configuration/python-agent-configuration.mdx:4057]: Just confirming the `.` is not a typo in `context` and `message`?

File: src/content/docs/apm/agents/python-agent/python-agent-api/recordlogevent-python-agent-api.mdx [hunk 3]
```diff
@@ -131,3 +147,13 @@ def fetch():
     newrelic.agent.record_log_event('Fetching data.')
     # do some type of work in this transaction...
 ```
+
+### Record log event with context data attributes [#context-data-example]
+
+An example of recording a log event using message attributes and context attributes
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/python-agent/python-agent-api/recordlogevent-python-agent-api.mdx:153]:
```suggestion
Here's an example of recording a log event using message attributes and context attributes:
```

File: src/content/docs/release-notes/agent-release-notes/python-release-notes/python-agent-90500.mdx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+version: 9.5.0
+downloadLink: 'https://pypi.python.org/pypi/newrelic'
+features: ['Add context data attributes to logs','Add support for dictionary type log messages','Obfuscate license keys and user API keys in logs','Add attribute support to `record_log_event` API']
+bugs: []
+security: []
+---
+
+## Notes
+
+This release of the Python agent adds context data attributes to logs, support for dictionary type log messages, adds attribute support to `record_log_event` API, and obfuscation of logged license keys and user API keys.
```
Comment (ally-sassman) [inline src/content/docs/release-notes/agent-release-notes/python-release-notes/python-agent-90500.mdx:13]:
```suggestion
This release of the Python agent adds context data attributes to logs, support for dictionary type log messages, adds attribute support to `record_log_event` API, and obfuscates logged license keys and user API keys.
```

File: src/content/docs/release-notes/agent-release-notes/python-release-notes/python-agent-90500.mdx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+## New features
+
+* **Add context data attributes to logs**
+Context data available from logging frameworks can now be recorded as attributes on logs. These attributes are prefixed with `context.`, which is not considered when filtering attributes from logs.
+
+Context data may include extras, dictionary message attributes, and attributes provided by logging frameworks.
+
+To report these attributes, [enable context data forwarding](/docs/agents/python-agent/configuration/python-agent-configuration#application_logging.forwarding.context_data.enabled) and optionally configure [include](/docs/agents/python-agent/configuration/python-agent-configuration#application_logging.forwarding.context_data.include) and [exclude](/docs/agents/python-agent/configuration/python-agent-configuration#application_logging.forwarding.context_data.exclude) rules to filter the allowed attributes.
```
Comment (ally-sassman) [inline src/content/docs/release-notes/agent-release-notes/python-release-notes/python-agent-90500.mdx:25]:
```suggestion
* **Add context data attributes to logs**
    * Context data available from logging frameworks can now be recorded as attributes on logs. These attributes are prefixed with `context.`, which is not considered when filtering attributes from logs.
    * Context data may include extras, dictionary message attributes, and attributes provided by logging frameworks.
    * To report these attributes, [enable context data forwarding](/docs/agents/python-agent/configuration/python-agent-configuration#application_logging.forwarding.context_data.enabled) and optionally configure [include](/docs/agents/python-agent/configuration/python-agent-configuration#application_logging.forwarding.context_data.include) and [exclude](/docs/agents/python-agent/configuration/python-agent-configuration#application_logging.forwarding.context_data.exclude) rules to filter the allowed attributes.
```

File: src/content/docs/release-notes/agent-release-notes/python-release-notes/python-agent-90500.mdx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+* **Add support for dictionary type log messages**
+Dictionary type log messages are now supported for logging frameworks which support dictionary type log messages (including the builtin `logging` module and `structlog`). The key `message` will be extracted if available and treated as the log message, and any other items will be considered context data attributes and prefixed with `message.`. These attributes are subject to the same filtering rules as other context data.
+
+* **Add attribute support to `record_log_event` API**
+The `record_log_event` API now allows passing context data in using the optional `attributes` keyword argument, and allows the message argument to be a dictionary as described above.
+
+* **Obfuscate license keys and user API keys in logs**
+The agent now obfuscates all instances of license keys or user API keys in agent logs and audit logs.
```
Comment (ally-sassman) [inline src/content/docs/release-notes/agent-release-notes/python-release-notes/python-agent-90500.mdx:34]:
```suggestion
* **Add support for dictionary type log messages**

    * Dictionary type log messages are now supported for logging frameworks which support dictionary type log messages (including the builtin `logging` module and `structlog`). 
    * The key `message` will be extracted if available and treated as the log message, and any other items will be considered context data attributes and prefixed with `message.`. These attributes are subject to the same filtering rules as other context data.

* **Add attribute support to `record_log_event` API**

    * The `record_log_event` API now allows passing context data in using the optional `attributes` keyword argument, and allows the message argument to be a dictionary as described above.

* **Obfuscate license keys and user API keys in logs**

    * The agent now obfuscates all instances of license keys or user API keys in agent logs and audit logs.
```

**Review by ghost — CHANGES_REQUESTED** [review]
See my suggestions.

**Review by ghost — APPROVED** [review]

**Pull request merged**